### ISSUE TITLE
Random output for markdown xml snippet

### ISSUE THREAD

**Issue by Dschaehn** [doxygen/doxygen#8322]
**Describe the bug**
When including the following xml code snippet in some .md file I get random output, which makes our XML parser crash:
```
~~~{xml}
<!--
Id="" Test
-->
~~~
```
I get e.g. this (output changes after each run):
```
<para><programlisting filename=".xml"><codeline><highlight class="comment">D</highlight></codeline>
<codeline><highlight class="comment">HQ]Ϋφ</highlight></codeline>
<codeline><highlight class="comment">--&gt;</highlight></codeline>
</programlisting> </para>
```

**Expected behavior**
It starts being instable in commit 0fc06d657d596adcc289a5f228973ea268efd66d. Before the output was stable and looked like this:
```
<para><programlisting filename=".xml"><codeline><highlight class="comment">&lt;!--</highlight></codeline>
<codeline><highlight class="comment">Id=&quot;&quot;<sp/>Test</highlight></codeline>
<codeline><highlight class="comment">--&gt;</highlight></codeline>
</programlisting> </para>
```
Even though the code snippet looks meaningless, it should at least produce a stable output without invalid characters. 

**To Reproduce**
Run doxygen with [instable_xml_snippet.zip](https://github.com/doxygen/doxygen/files/5796234/instable_xml_snippet.zip)

**Version**
Windows 10, 64 bit, version 1.9.1 (ef9b20ac7f8a8621fcfc299f8bd0b80422390f4b), downloaded from https://doxygen.nl/files/doxygen-1.9.1-setup.exe

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #8320 

(please also mention always the, output, file when you show a snippet, makes searching a bit easier. This times there were not that many files but still).

**Comment by albert-github** [comment]
Code has been integrated in master on github (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.2.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
